src/mscorlib/shared/System/IO/PathInternal.cs

```diff
@@ -10,8 +10,12 @@ internal static partial class PathInternal
         /// <summary>
         /// Returns true if the path ends in a directory separator.
         /// </summary>
-        internal static bool EndsInDirectorySeparator(string path) =>
-            !string.IsNullOrEmpty(path) && IsDirectorySeparator(path[path.Length - 1]);
+        internal static bool EndsInDirectorySeparator(ReadOnlySpan<char> path) => path.Length > 0 && IsDirectorySeparator(path[path.Length - 1]);
```

Comment:
We seem to only call these methods when path.Length is greater than 0 anyway. Can the length check be a Debug.Assert instead?

Same with StartsWithDirectorySeparator.

Reply:
That is true now, but I expect that to change, particularly as I expect to include this file in a few CoreFX libraries.

Reply:
We seem to have EndsInDirectorySeparator in already:
https://github.com/dotnet/coreclr/blob/22f1bc00d018a49f9550ee3b564f5f7737960b0d/src/mscorlib/shared/System/IO/Path.Windows.cs#L221

But not in Path.Unix.cs (and we don't have StartsWithDirectorySeparator)

Reply:
Yes, missed that. I'll get this one. I have a fair amount of cleanup to do- I intend to do so when I include the Path sources directly in System.Runtime.Extensions again.